packages/core/src/browser/status-bar/status-bar.ts

```diff
@@ -63,6 +64,7 @@ export interface StatusBar {
     setBackgroundColor(color?: string): Promise<void>;
     setElement(id: string, entry: StatusBarEntry): Promise<void>;
     removeElement(id: string): Promise<void>;
+    readonly onStatusBarEntriesLoaded: Event<boolean>;
```

Comment:
why do we need it? Could you have a look like displaying language in the status bar is done?

Reply:
I am afraid not :) The "displaying language" in the status bar is a not-clickable label, while the "Spaces: 4" is a clickable button. If I don't remove the button when theia loads and the user clicks it, the front-end would throw an error that says the command is not run-able (due to not having any active editor in the shell).

Reply:
the root cause is that current and active editors are not set properly after reload and it should be fixed, please remove changes to the status bar